.github/workflows/checks.yml

```
@@ -0,0 +1,35 @@
+name: Checks
+
+on:
+  pull_request:
+
+jobs:
+
+  lint-job:
+    name: Linting and Type checks (Python-${{ matrix.python-version }})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.10", "3.11", "3.12"]
```

```suggestion
        python-version: ["3.12"]
```
Is it necessary to lint with all these python versions?

---

There might be subtle changes among the different Python versions. I used the `BucketFS-Python` Project as example (see https://github.com/exasol/bucketfs-python/blob/c9e203dcf86c762997a4d67903b0e02091afcc06/.github/workflows/checks.yml#L28) , which I thing created all the github actions from the `exasol-toolbox`. So I guess this is kind of our standard. Maybe @tkilias can add something.